airflow/providers/amazon/aws/sensors/batch.py

```diff
@@ -74,16 +73,13 @@ def poke(self, context: Context) -> bool:
 
         raise AirflowException(f"Batch sensor failed. Unknown AWS Batch job status: {state}")
 
```

Comment:
Do we need to deprecate the get_hook() method?

Reply:
Yeah, better deprecate and return `self.hook`

Reply:
I was thinking about it, there is the "every change is a breaking change" way of thinking, but also I don't see why anyone would go out of their way to get the hook from a sensor outside of that sensor...

Reply:
One use case it create custom Operator/Sensor which based on existed Sensor/Operator and overwrite particular method.

So better to keep existed public method at least for a while. 

And we need to start remove deprecated part again in next major release.